rasa/engine/graph.py

```diff
@@ -194,5 +249,48 @@ def from_schema_node(
             fn_name=schema_node.fn,
             inputs=schema_node.needs,
             eager=schema_node.eager,
+            model_storage=model_storage,
             execution_context=execution_context,
+            resource_name=schema_node.resource_name,
+        )
+
+
+def graph_schema_as_dict(graph_schema: GraphSchema) -> Dict[Text, Any]:
+    """Returns graph schema in a serializable format.
+
+    Args:
+        graph_schema: The graph schema to serialize.
+
+    Returns:
+        The graph schema in a format which can be dumped as JSON or other formats.
+    """
+    serializable_graph_schema = {}
+    for node_name, node in graph_schema.items():
+        serializable = dataclasses.asdict(node)
+
+        # Classes are not JSON serializable (surprise)
+        serializable["uses"] = f"{node.uses.__module__}.{node.uses.__name__}"
+
+        serializable_graph_schema[node_name] = serializable
+
+    return serializable_graph_schema
+
+
+def load_graph_schema(serialized_graph_schema: Dict[Text, Any]) -> GraphSchema:
+    """Loads a graph schema which has been serialized using `graph_schema_as_dict`.
+
+    Args:
+        serialized_graph_schema: A serialized graph schema.
+
+    Returns:
+        A properly loaded schema.
+    """
+    graph_schema = {}
+    for node_name, serialized_node in serialized_graph_schema.items():
+        # TODO: Test and handle error here
+        serialized_node["uses"] = rasa.shared.utils.common.class_from_module_path(
+            serialized_node["uses"]
         )
+        graph_schema[node_name] = SchemaNode(**serialized_node)
+
+    return graph_schema
```

Comment:
@joejuzl Quickly added these here. They need some more testing and they are also causing circular imports in `model_storage` which doesn't feel like they are necessarily in the right place.